blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DBasicMetricsRW.java

```diff
@@ -0,0 +1,350 @@
+    private Map<Locator, MetricData> getStringDataForRange( List<Locator> locators, Range range ) {
+
+        return getStringBooleanDataForRange( locators, range, false );
+    }
+
+    /**
+     * Return map of MetricDatas for strings or booleans in the given range, depending on the isBoolean param.
+     *
+     * @param locators
+     * @param range
+     * @param isBoolean if true, return MetricDatas as type Boolean
+     * @return
+     */
+    private Map<Locator, MetricData> getStringBooleanDataForRange( List<Locator> locators, Range range, boolean isBoolean ) {
+
+        Map<Locator, MetricData> metrics = new HashMap<Locator, MetricData>();
+        Map<Locator, ResultSetFuture> futures = new HashMap<Locator, ResultSetFuture>();
+
+        for( Locator locator : locators ) {
+            futures.put( locator, rawIO.getStringAsync( locator, range ) );
+        }
+
+        for( Map.Entry<Locator, ResultSetFuture> future : futures.entrySet() ) {
+
+            Points points = new Points<String>();
+
+            try {
+                for ( Row row : future.getValue().getUninterruptibly().all() ) {
+
+                    Object value = isBoolean ? Boolean.valueOf( row.getString( VALUE ) ) : row.getString( VALUE );
+                    points.add( new Points.Point( row.getLong( COLUMN1 ), value ) );
+                }
+
+                MetricData.Type type = isBoolean ? MetricData.Type.BOOLEAN : MetricData.Type.BOOLEAN;
+
+                metrics.put( future.getKey(), new MetricData( points, getUnitString( future.getKey() ), type ) );
+            }
+            catch (Exception e ) {
+
+                LOG.error( String.format( "error writing metric %s, metrics_string", future.getKey() ), e );
+            }
+        }
+
+        return metrics;
+    }
+
+
+    /**
+     * Return map of MetricDatas for boolean values in the given range.
+     *
+     *
+     * @param locators
+     * @param range
+     * @return
+     */
+    private Map<Locator, MetricData> getBooleanDataForRange( List<Locator> locators, Range range ) {
+
+        return getStringBooleanDataForRange( locators, range, true );
+    }
+
+    /**
+     * When the type is not known for a list of locators, do the following:
+     * 1) first try to get them as numerics.
+     * 2) If that call is empty, see if they are in metrics_string (ignore granularity & columnFamily)
+     *
+     * @param locators
+     * @param range
+     * @param columnFamily
+     * @param gran
+     * @return
+     * @throws IOException
+     */
+    private Map<Locator, MetricData> getNumericOrStringDataForRange( List<Locator> locators,
+                                                                     Range range,
+                                                                     String columnFamily,
+                                                                     Granularity gran ) throws IOException {
+
+        Map<Locator, MetricData> metrics = new HashMap<Locator, MetricData>();
```

Comment:
In AstyanaxReader, there is a call to ```Instrumentation.markScanAllColumnFamilies()``` when this method is called. Do we want to preserve that?